### ISSUE TITLE
CheckHealthAsync is behaving differently in different scenarios 

### ISSUE THREAD

**Issue by AvtsVivek** [dapr/dotnet-sdk#840]
I have a very simple .NET 6 console app to understand dapr state store based on the [dapr getting started ms docs here](https://docs.microsoft.com/en-us/dotnet/architecture/dapr-for-net-developers/getting-started#add-dapr-state-management).

You can see the [full sample here](https://github.com/AvtsVivek/DaprIssues/tree/main/src/DaprCounter).

As you can see in [my Program.cs](https://github.com/AvtsVivek/DaprIssues/blob/4a43804b9e150d27dbbf05cd86abd047f8c9bc8e/src/DaprCounter/Program.cs#L14) file, I call the [dapr client's CheckHealthAsync method](https://github.com/dapr/dotnet-sdk/blob/a6319dd4b6db0884f5b2cc325dfa4c8bed06e63c/src/Dapr.Client/DaprClient.cs#L315) to determine the health. 

When I run the app in the dapr context using the following command, the health check call returns true meaning its healthy.

```ps
dapr run --app-id DaprCounter dotnet run
```

![image](https://user-images.githubusercontent.com/41255281/154843951-772bd2fc-8b41-4374-9d68-74a5a93572ad.png)

For the other scenario, I opened [this project folder in vs code](https://github.com/AvtsVivek/DaprIssues/tree/main/src/DaprCounter). 

I have added a couple of launch configurations.

The first one is based on [omni sharp](https://github.com/OmniSharp/omnisharp-vscode) as follows.

![image](https://user-images.githubusercontent.com/41255281/154843087-9841b6d5-2e54-47f9-96c5-c5f4e8b4da79.png)

Then on top if it, I added another launch configuration based on [vs code dapr extension](https://docs.dapr.io/developing-applications/ides/vscode/vscode-dapr-extension/). 

![image](https://user-images.githubusercontent.com/41255281/154843187-32a2eaca-4853-4095-b593-9b1323719211.png)

The resulting [launch.json and tasks.json are here in the .vscode folder](https://github.com/AvtsVivek/DaprIssues/tree/main/src/DaprCounter/.vscode) for you to see.

I execute the dapr launch configuration as follows.

![image](https://user-images.githubusercontent.com/41255281/154843477-f5b62c2b-66e3-4693-b9ae-3b2d29b08ba7.png)

This time around I see that the health check call returns **false** meaning its **NOT** healthy.

![image](https://user-images.githubusercontent.com/41255281/154843789-6cddde5a-7255-4961-8b86-44d826a1633e.png)

So in summary, when I run the app using vs code launch configuration, the healthcheck returns false. But when I run the app using the  dapr cli it returns true. So what am I missing. Is this by design? How to understand this?

**Comment by rynowak** [comment]
@philliphoff - do you have any visibility into what might be different in these cases?

The SDK uses GRPC for the state store and HTTP for the health checks. So if the HTTP port is not configured correctly via env-vars then that would explain this behavior.

**Comment by rynowak** [comment]
Another explanation is that makes sense...

The sidecar exposes two health endpoints (inbound and outbound)

- Inbound is healthy when your app can recieve traffic from other sidecars
- Outbound is healthy when your app can send traffic to dapr

I suspect what's happening here is a misconfiguration in your vscode files: https://github.com/AvtsVivek/DaprIssues/blob/main/src/DaprCounter/.vscode/tasks.json#L42

That file configures a listening port (5000) but your sample doesn't listen for incoming requests at all. I suspect if you remove that line from `tasks.json` that will resolve the issue you're seeing.

@halspang is working on adding another SDK api for checking outbound health, which is probably what you're looking for.

**Comment by AvtsVivek** [comment]
Yes, thanks. 
When I removed that line, its showing healthy.
